### DISCUSSION TITLE
is there a way to have custom commands when displaying the prompt

### DISCUSSION THREAD

**Discussion by Xiaorui-Huang** [JanDeDobbeleer/oh-my-posh#1706] [Troubleshoot] [answered]
Before using oh-my-posh I added the functionality to save the current working directory every time the prompt is displayed using the prompt function in Powershell. This is so that pwsh will cd into the last working directory directly if I open another pwsh instance.

    function prompt{  
        #Tracking the current working directory
        if ( "$PWD" -ne "$recorded_cwd"){
            $global:recorded_cwd = $PWD
            echo "$PWD" > ~\"$env:USERNAME-cwd.txt"
        }
    }
    # Current working directory persistency (also see prompt function)
    if ((Test-Path ~\"$env:USERNAME-cwd.txt") -and ("$PWD" -eq "$HOME")){
        cd $(gc ~\"$env:USERNAME-cwd.txt" -head 1)
    }
however, calling `oh-my-posh --init ...` seems to have replaced the prompt function and my current wording directory won't get saved upon prompt display.

Is there a way to configure some custom commands in the configuration file of oh-my-posh?

**Comment by JanDeDobbeleer** [comment, marked as answer]
You can override the `Set-PoshContext` function to have the same effect.

**Reply by Kipre** [reply]
How about prepending something to the prompt itself, like [here](https://docs.microsoft.com/en-us/windows/terminal/tutorials/new-tab-same-directory) for opening new tabs in the same directory ?

**Reply by JanDeDobbeleer** [reply]
@kipre that's already available using `"osc99": true` in your config.

**Reply by Kipre** [reply]
This is great, thanks a lot !
